lib/zlib.js

```diff
@@ -156,6 +156,56 @@ function flushCallback(level, strategy, callback) {
   }
 }
 
+// 1. Returns true for finite numbers
+// 2. Returns false for undefined and NaN
+// 3. Throws ERR_INVALID_ARG_TYPE for non-numbers
+// 4. Throws ERR_OUT_OF_RANGE for infinite numbers
+function checkFiniteNumber(number, name) {
+  if (Number.isFinite(number)) {
+    return true;  // is a valid number
+  } else {
+    // undefined or NaN
+    if (number === undefined || Number.isNaN(number)) {
+      return false;
+    }
```

Comment:
I would check for `undefined` first because it is the cheapest check and it is the most likely one (I guess the common use case is to rely on defaults).

So:

```js
if (number === undefined)
  return false;
if (Number.isFinite(number)
  return true;
if (Number.isNaN(number)
  return false;

...
```